src/applications/pact-act/utilities/display-logic-results.js

```diff
@@ -58,13 +59,42 @@ export const responsesMatchResultsDCs = (yesShortNames, pageDCsForFlow) => {
  */
 export const determineResultsPage = (formResponses, router) => {
   const responseToServicePeriod = getServicePeriodResponse(formResponses);
+  const { RESULTS_4 } = SHORT_NAME_MAP;
 
-  const allResultsDCs = DISPLAY_CONDITIONS?.RESULTS;
+  const resultsPages = DISPLAY_CONDITIONS?.RESULTS;
   const yesShortNames = filterForYesResponses(formResponses);
 
-  for (const resultsPage of Object?.keys(allResultsDCs)) {
+  // Because Results 4 requires answers to specific questions rather than any answer in a batch,
+  // it must be evaluated first when determining the results page as other results pages'
+  // display conditions may also be true
+  const pageDCsForResults4 =
```

Comment:
Adding special handling for results page 4 (see comment right above this for why it has to be evaluated in a special way).